### PR TITLE
Feature/173 implement localdev docker

### PR DESCRIPTION
**Pull request by adamhgriffith-uofu**
Added containerized method for building artifacts on a developer's machine. Updated the `README.md` file accordingly.
If this is useful for folks and works may we merge with `master`?

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1 +1,5 @@
-build/
+# IDE Metadata
+.vs/
```
Comment (sthapa) [inline .gitignore:2]: Can you remove the ide specific ignores.  I think the best practice is to add this to your global gitignore file and not to pollute project specific gitignore with these directories

**Comment by sthapa** [comment]
I tried building and running the container and got an error: `/docker-entrypoint.sh: line 7: cd: /work/build: No such file or directory`
 I don't think your Dockerfile creates the /work/build dir, or at the very least I couldn't find it.

File: .gitignore (hunk 1 above)
Comment (adamhgriffith-uofu) [reply .gitignore]: IDE Metadata section is now removed from the `.gitignore`.

**Comment by adamhgriffith-uofu** [comment]
> I tried building and running the container and got an error: `/docker-entrypoint.sh: line 7: cd: /work/build: No such file or directory` I don't think your Dockerfile creates the /work/build dir, or at the very least I couldn't find it.

When you ran the container was `-v ${PWD}:/work` included?

**Comment by adamhgriffith-uofu** [comment]
I will admit my primary interest lies with the `slate-service` build artifact (compiled under Centos7). Eventually we could re-write the disabled Jenkins pipeline to use the same container (one vetted build environment for both server development and deployment).

**Comment by sthapa** [comment]
I'd see the usefulness in not being tied to having a Centos 7 host for building.  E.g. I can build the code on my fedora workstation or the rocky 8 build cluster and then copy the binaries over to the production machine.  Maybe a bit niche, but I'd prefer not to use the production server for also building the binaries.

**Review by sthapa — APPROVED** [review]
LGTM

**Comment by cnweaver** [comment]
Okay. I agree that using Docker can be a good way to get buils done for target systems without having to have full installs of them. One thing I would note is that it's a kind of important thing long-term *not* to have a single blessed development environment, as that nearly always leads to portability and code quality suffering. You don't want to end up with a situation in which after years of development the codebase only works on one distro with the bugs particular to it and its single version of gcc. That being said, it makes a lot of sense to keep an eye on important, officially supported platforms on which the code must work (which list is undoubtedly headed by CentOS 7 in this case).

**Pull request merged**